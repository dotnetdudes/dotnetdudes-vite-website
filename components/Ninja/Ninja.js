import NinjaTemplate from "./template";

// <Ninja></Ninja>
class Ninja extends HTMLElement {
  constructor() {
    super();
  }
  connectedCallback() {
    this.innerHTML = NinjaTemplate;
  }
  disconnectedCallback() {
    //
  }
  attributeChangedCallback(attrName, oldVal, newVal) {
    //
  }
}

window.customElements.define('css-ninja', Ninja);