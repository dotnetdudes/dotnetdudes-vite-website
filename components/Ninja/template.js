const NinjaTemplate = /* html */ `
<span id="ninja">
    <!-- CSS Ninja -->
    <div class="ninja green">
        <div class="n--head">
            <div class="n--ribbon"></div>
            <div class="n--eyes"></div>
        </div>
        <div class="n--body">
            <div class="n--weapon"></div>
            <div class="n--arm"></div>
            <div class="n--arm"></div>
            <div class="n--leg"></div>
            <div class="n--leg"></div>
        </div>
    </div>
    <blockquote class="oval-thought-border">
        <h4 class="bubtext">
            Develop your potential with dotnetdudes.
        </h4>
        <p>Enquire about working together!</p>
    </blockquote>
    <div class="small-heading">
        <h4 class="bubtext">
            Develop your full potential with dotnetdudes.
        </h4>
        <p>Enquire about working together!</p>
    </div>
</span>
`;

export default NinjaTemplate;