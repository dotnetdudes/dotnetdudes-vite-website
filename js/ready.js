// import { getInitialTheme, setTheme } from './theme';

export const onSiteLoaded = () => {
    try {

        document.addEventListener('DOMContentLoaded', function () {
            // const theme = getInitialTheme();
            // console.log(theme);
            // setTheme(theme);
            // const pnl = document.getElementById('loadingpanel');
            // pnl.style.display = 'none';
            console.log('DOMContentLoaded');
        });
    } catch (error) {
        console.log(error);
    }
};
