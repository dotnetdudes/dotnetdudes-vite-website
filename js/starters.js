export const onStarterLoaded = () => {
    try {
        const openReactStarter = () => {
            window.open("https://github.com/dotnetdudes/react-starter", "_blank");
        }
        const openReactNativeStarter = () => {
            window.open("https://github.com/dotnetdudes/react-native-starter", "_blank");
        }
        document.addEventListener('DOMContentLoaded', function () {
            const btn = document.getElementById('reactStarter');
            const reactBtn = document.getElementById('nativeStarter');
            btn.addEventListener('click', openReactStarter);
            reactBtn.addEventListener('click', openReactNativeStarter);
        });
    } catch (error) {
        console.log(error);
    }
}
