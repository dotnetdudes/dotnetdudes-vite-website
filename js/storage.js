export const setItem = (name, item) => {
    if (!window.localStorage) return false;
    window.localStorage.setItem(name, item);
    return true;
};

export const getItem = (name) => {
    if (!window.localStorage) return false;
    return window.localStorage.getItem(name);
};

export const clearStorage = () => {
    if (!window.localStorage) return false;
    window.localStorage.clear();
};
