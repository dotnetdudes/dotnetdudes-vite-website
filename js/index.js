import { onSiteLoaded } from './ready';
/* import dotenv from 'dotenv';
const dotenv_result = dotenv.config({ silent: true }); */

// onSiteLoaded();
console.log('okay');
