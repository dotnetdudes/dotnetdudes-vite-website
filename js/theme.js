import { getItem, setItem } from "./storage";

export const isDarkMode = () =>
  window.matchMedia &&
  window.matchMedia("(prefers-color-scheme: dark)").matches;

export const getInitialTheme = () => {
  let initialTheme = isDarkMode() ? "dark" : "light";
  // console.log(isDarkMode());
  if (getItem("dudesTheme")) initialTheme = getItem("dudesTheme");
  return initialTheme;
};

export const setTheme = (theme) => {
  // set css link href
  setItem("dudesTheme", theme);
};
