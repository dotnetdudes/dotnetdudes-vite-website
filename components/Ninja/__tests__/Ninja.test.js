import { test, expect, describe } from "vitest";
import "../Ninja.js";

describe("Ninja static css", () => {
  
});