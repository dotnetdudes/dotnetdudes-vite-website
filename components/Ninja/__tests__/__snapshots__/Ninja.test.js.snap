// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ninja static css > should match the current snapshot 1`] = `""`;
